lib/apiservers/engine/backends/container.go

```diff
@@ -549,9 +559,7 @@ func (c *Container) containerStart(name string, hostConfig *container.HostConfig
 	if bind {
 		e := c.findPortBoundNetworkEndpoint(hostConfig, endpoints)
 		if err = c.mapPorts(portmap.Map, hostConfig, e); err != nil {
-			err = fmt.Errorf("error mapping ports: %s", err)
-			log.Error(err)
-			return InternalServerError(err.Error())
+			return InternalServerError(fmt.Sprintf("error mapping ports: %s", err))
```

Comment:
Why no logging here - even if mapPorts logs the initial error I don't think it has the requisite knowledge to record which container had the problem.
Admittedly the original code didn't add that useful bit of data either, and the context based logging @fdawg4l has in WIP would address it in a cleaner fashion.